entry/src/entry.rs

```diff
@@ -111,15 +111,22 @@ pub struct Api<'a> {
 /// field points to Transactions that took place shortly before `hash` was generated.
 ///
 /// If you multiply `num_hashes` by the amount of time it takes to generate a new hash, you
-/// get a duration estimate since the last Entry. Since processing power increases
+/// get a duration estimate since the last `Entry`. Since processing power increases
 /// over time, one should expect the duration `num_hashes` represents to decrease proportionally.
 /// An upper bound on Duration can be estimated by assuming each hash was generated by the
 /// world's fastest processor at the time the entry was recorded. Or said another way, it
 /// is physically not possible for a shorter duration to have occurred if one assumes the
 /// hash was computed by the world's fastest processor at that time. The hash chain is both
 /// a Verifiable Delay Function (VDF) and a Proof of Work (not to be confused with Proof of
 /// Work consensus!)
-
+///
+/// `Entry` contains `transactions` that are executable in parallel.  So an `Entry` is a unit of
+/// parallelism at the chain level.  All transactions in the `transactions` field have to follow the
```

Comment:
```suggestion
/// parallelism within a block.  All transactions in the `transactions` field have to follow the
```

Reply:
I was trying to point out that parallelism in Solana is expressed by Entries.
The scope is smaller than a block - it is a single Entry.
But still, it is what allows the validators to run transactions in parallel.
Ethereum and EVM do not have this property.  All transactions are computed sequentially.

So, saying `within a block` seems wrong.  Unless I am misunderstanding how this parallelism works.  See my next response.

Reply:
> The scope is smaller than a block - it is a single Entry.

When I read this, I interpreted "at the chain level" to have larger scope. "chain" is shorthand for "blockchain" which implies multiple blocks. So maybe I misinterpreted what you meant here?

> But still, it is what allows the validators to run transactions in parallel.

Slightly off topic, but I'd say entries aid in allowing parallel tx processing, but in code, you'll see that we attempt to execute multiple entries in parallel if we can (see the `if` statement at line 508 below):
https://github.com/solana-labs/solana/blob/2da02992b79eec57095e4aa4010495310be86367/ledger/src/blockstore_processor.rs#L460-L552

At the end of it, we execute transactions in parallel, and entries are known, small chunks that can be parallelized.

> 1) `Entry` contains `transactions` that are executable in parallel.
> 2) So an `Entry` is a unit of parallelism at the chain level.

I think I maybe see what you were driving at now. How about if swapped out "transaction" for "chain" ? I think "chain" is ambiguous / open to interpretation. But also, re-reading, I'm not sure that sentence 2) adds anything beyond sentence 1) so ripping it out is also an option IMO

Reply:
Transaction in an `Entry` are required to be executable in parallel.
As referenced here:

https://github.com/solana-labs/solana/blob/2da02992b79eec57095e4aa4010495310be86367/ledger/src/blockstore_processor.rs#L519-L534

While parallel execution of transactions across multiple `Entry`es is an optimization.

Maybe I am overthinking this requirement.  But it is the reason for `Entry` existence, according to my current understanding.
And this is why I wanted to spell it out.
The second sentence explains why `Entry`es are there at all.
The first sentence leads to it, explaining the mechanics.

> I think I maybe see what you were driving at now. How about if swapped out "transaction" for "chain" ? I think "chain" is ambiguous / open to interpretation.

Why “chain” is ambiguous?  Wouldn't it mean the Solana chain in the context we are in?

There seems to be two mechanisms, at the moment, that express transaction parallelism in Solana: `Entry`es and explicit account access lists.

And the validator code ensures they match.
I can imagine either of them removed:
1. `Entry`es are removed, putting transactions directly into blocks.
   Explicit account access lists still allow for parallel execution.
2. Explicit account access lists are removed.
  `Entry`es express the desire for parallel execution.
  When transactions are executed, account access lists are reconstructed as part of transaction execution.  Providing the same checks as are statically ensured right now.